webrender/src/renderer.rs

```diff
@@ -1481,6 +1517,93 @@ impl Renderer {
                 ResultMsg::RefreshShader(path) => {
                     self.pending_shader_updates.push(path);
                 }
+                ResultMsg::DebugCommand(command) => {
+                    self.handle_debug_command(command);
+                }
+            }
+        }
+    }
+
+    #[cfg(not(feature = "debugger"))]
+    fn update_debug_server(&self) {
+    }
+
+    #[cfg(feature = "debugger")]
+    fn update_debug_server(&self) {
+        while let Ok(msg) = self._debug_server.debug_rx.try_recv() {
+            match msg {
+                DebugMsg::FetchBatches(sender) => {
+                    let mut batch_list = BatchList::new();
+
+                    if let Some(ref frame) = self.current_frame {
+                        if let Some(ref frame) = frame.frame {
+                            for pass in &frame.passes {
+                                for target in &pass.alpha_targets.targets {
+                                    batch_list.push("[Clip] Clear", target.clip_batcher.border_clears.len());
```

Comment:
Definitely - the batch list is very much a proof of concept - I'll expand on the information with a more detailed view as a follow up.